File: apple/Sources/ReactNativeWatchOS/Modifiers/AccessibilityLabel.swift
import SwiftUI

/// SwiftUI `.accessibilityLabel(_:)`. Overrides the label VoiceOver reads to
/// identify the view.
enum RNWAccessibilityLabelModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("accessibilityLabel") { view, params, _ in
            AnyView(view.accessibilityLabel(Text(params.string("label") ?? "")))
        }
    }
}

File: src/modifiers/accessibilityLabel.ts
import { createModifier } from './createModifier';

/// SwiftUI `.accessibilityLabel(_:)`. Sets the label VoiceOver reads to
/// identify the view, overriding any inferred label.
export function accessibilityLabel(label: string) {
  return createModifier('accessibilityLabel', { label });
}
